### PR TITLE
Feature/accept

### PR DESCRIPTION
**Pull request by ninjarails**
Working on Reddot, we need to make the customer apps PCI compliance. Authorize.net has a feature called Accept.js. This PR has commits for backend processing of Accept.js. https://developer.authorize.net/api/reference/features/accept.html

### CODE REVIEW

File: app/models/accounting/payment.rb [hunk 1]
```diff
@@ -23,25 +25,27 @@ class Payment < ::ActiveRecord::Base
...
 
     validates :account_type, inclusion: { in: Accounting::Payment.account_types.keys }, if: :ach?
 
-    validates :number, length: { in: 13..16 }, if: :card?
+    validates :number, length: { in: 13..16 }, if: :card?, unless: :accept?
+
+    validates :number, :ccv, presence: true, if: :card?, unless: :accept?
 
-    validates :number, :ccv, :month, :year, presence: true, if: :card?
+    validates :number, :ccv, numericality: { only_integer: true }, if: :card?, unless: :accept?
```
Comment (alexbrinkman) [inline app/models/accounting/payment.rb]: @ninjarails - I don't think we should still have fields for the number or the ccv.  I could see the number, if it's only the last four of the card.  I don't know why we'd ever have ccv.  Aren't we just getting the token or sconce back from Authorize to store?

File: app/services/accounting/accept_service.rb [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        @response = transaction.create_customer_payment_profile(build_request)
+
+        if @response != nil
+          if @response.messages.resultCode == MessageTypeEnum::Ok
+            Accounting.log 'Payment', 'Accept', info: "Successfully created a customer payment profile with id: #{@response.customerPaymentProfileId}."
+            @payment.assign_attributes(
+              title: parsed_response[:card_type],
+              payment_profile_id: @response.customerPaymentProfileId,
+              default: @profile.payments.count == 0,
+              last_four: parsed_response[:account_number].to_s[-4..-1]
```
Comment (alexbrinkman) [inline app/services/accounting/accept_service.rb]: This makes me think we are getting the full card number at this point, is that true?

Comment (ninjarails) [reply app/services/accounting/accept_service.rb]: No, API returns masked card number like: xxxxx1234.

File: app/models/accounting/payment.rb (hunk 1 above)
Comment (ninjarails) [reply app/models/accounting/payment.rb]: I don't actually use some of these validations for Accept.js. But it's used when CSR creates payment method.

**Pull request merged**